File: scratch/inspect_test_output.dart
import 'dart:convert';
import 'dart:io';

void main() {
  final content = File('test_output.pdf').readAsStringSync(encoding: latin1);
  for (final chunk in content.split('endobj')) {
    if (chunk.contains('/Link') || chunk.contains('/URI')) {
      print('--- OBJ ---');
      print(chunk.trim());
    }
  }
}

File: scratch/inspect_test_output.py
import sys

with open('test_output.pdf', 'rb') as f:
    content = f.read().decode('latin-1')

print("PDF Objects containing /Link or /URI:")
for chunk in content.split('endobj'):
    if '/Link' in chunk or '/URI' in chunk:
        print("--- OBJ ---")
        print(chunk.strip())
